.github/workflows/test-helpers/wait-for-issue

```diff
@@ -1,8 +1,23 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-function download-and-extract {
-  issue_number="$1"
+function download-and-extract-issue {
+  local issue_number="$1"
+
+  # I tried `gh issue view` but it does whack stuff with the terminal that makes
+  # it non-trivial to redirect/pipe. Pulling from JSON is probably more robust
+  # anyway.
+
+  gh api "/repos/:owner/:repo/issues/${issue_number}" > issue
+  jq -r .state issue > issue-state
+  jq -r .comments issue > issue-ncomments
+  gh api "/repos/:owner/:repo/issues/${issue_number}/comments" > issue-comments
+  jq -r '.[0].body' issue-comments > issue-comment
+}
```

Comment:
Can we not make this set the values as output so we simply consume the outputs of the `setup` step in downstream `test` steps?

Reply:
Good idea!

Reply:
Working this through ... I don't think I'm going to use GHA's `outputs` functionality, because I don't really trust it security-wise. It's probably okay, but I feel safer using files to communicate.

<img width="832" alt="Screen Shot 2021-02-17 at 10 24 02 AM" src="https://user-images.githubusercontent.com/134455/108226058-892d6f80-710a-11eb-9206-355a345c0bb2.png">

It's also more verbose:

```
          test "${{ steps.setup.outputs.issue_state }}" = "open"
          test "${{ steps.setup.outputs.issue_ncomments }}" = "0"
```

vs. 

```
          test "$(cat issue-state)" = "open"
          test "$(cat issue-ncomments)" = "0"
```

GHA's `output` functionality is such a gnarly hack to begin with, it's not like it's way more elegant than using files anyway.

That said, I do like the idea of pushing the `wait` part up into setup. 👍